### PR TITLE
chore: bump Github Actions versions, add Python 3.13 tests

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

Title says it all.

### Context of Change

Deprecation notices on Actions:
> The following actions use a deprecated Node.js version and will be forced to run on node20: actions/checkout@v3, actions/setup-node@v3, actions/cache@v3. For more info: https://github.blog/changelog/2024-03-07-github-actions-all-actions-will-run-on-node20-instead-of-node16-by-default/

CodeQL v2 will also be deprecated in December: https://github.blog/changelog/2024-01-12-code-scanning-deprecation-of-codeql-action-v2/

### Type of Change

- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### Did you update CHANGELOG.md?
- [ ] Yes
- [x] No, this change does not impact library users

## Test Plan

The deprecation notices went away.

## Summary by CodeRabbit

- **Chores**
	- Updated GitHub Actions workflow configurations for integration, snippet, and unit tests to utilize newer versions of actions, enhancing performance and reliability.
	- Expanded Python version support to include Python 3.13 across all workflows.

### CODE REVIEW

**Comment by mvadari** [comment]
@coderabbitai review

**Comment by mvadari** [comment]
https://github.com/coderabbitai review

**Review by pdp2121 — APPROVED** [review]

File: .github/workflows/integration_test.yml [hunk 1]
```diff
@@ -21,11 +21,11 @@ jobs:
```
Comment (ckeshava) [inline .github/workflows/integration_test.yml:21]: While we are on the topic of upgrading the dependencies, can you update the Poetry version on line 4?
Versions `1.8.4` (released 11 days ago) or version `1.8.3` (released 6 months ago) are suitable.

https://pypi.org/project/poetry/#history

Comment (ckeshava) [reply .github/workflows/integration_test.yml:21]: The poetry version is not up-to-date in the other two github workflow files either

**Review by ckeshava — APPROVED** [review]

**Pull request merged**